File: dev/modbus_scripts/napdos/common/port.c
/* Example of Read/Write Port for LinCon.

   Author: Moki Matsushima

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2, or (at your option)
   any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program; if not, write to the Free Software Foundation,
   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. */

/* File level history (record changes for this file here.)

   v 0.0.0  1 Sep 2004 by Moki Matsushima
     create, blah blah... */

#include <stdio.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <sys/ioctl.h>
#include <unistd.h>
#include "msw.h"

int main()
{
	int doVal,wRetVal;

	/* open device file */
	wRetVal = Open_Slot(SLOT1);
	
	if (wRetVal > 0) {
		printf("open Slot failed!\n");
		return (-1);
	}
	
	printf("Module ID = %d \n",IoRead(SLOT1,0x0));
	/* read, get out if error or ESC pressed */
	doVal = 1;
	
	while (getchar() != 27) {
		if(doVal < 0xFF) doVal=doVal*2;
		else doVal = 1;
		IoWrite(SLOT1,0x0,doVal);
	}
	puts("End of program");

	Close_Slot(SLOT1);
	return SUCCESS;
}

File: dev/modbus_scripts/napdos/java/slot/i8k/Port.java
/*	Port Input/Output DEMO for slot parallel module
 
   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2, or (at your option)
   any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program; if not, write to the Free Software Foundation,
   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.

   Port.java
 
 	
 	v 0.0 2005.4.1 by moki matsushima
 
 	Copyright (c) 2005 ICPDAS, Inc. All  Rights Reserved.
 */

import java.io.*;							         //For System.in.read()
import com.icpdas.slot.*;						   //Slot I/O packages

public class Port 
{
	public static void main(String[] args) throws java.io.IOException
	{
		int rev;
		long diValue;
		int i=1;
		byte a[] = new byte[100];
		Slot slot1 = new Slot();
		rev = slot1.open(1);						   //open slot 1 for port i/o
		if (rev!=0) System.out.println("Open port error code : "+rev);
		else{
			while(a[0]!=113) {
				rev = slot1.inb(1,0);
				if (rev!=0) System.out.println("Read from slot1 offset 0 : "+ rev);
				slot1.outb(1,0,i);
				System.out.println("Write to slot1 offset 0 : "+ i);
				System.in.read(a);
				i=(i>255)?1:(i<<1);
			}
		}
		slot1.close(1);
		System.out.println("End of program");		
	}
}

